xbmc/windowing/gbm/drm/DRMAtomic.cpp

```diff
@@ -149,6 +149,11 @@ void CDRMAtomic::DrmAtomicCommit(int fb_id, int flags, bool rendered, bool video
   {
     CLog::Log(LOGERROR, "CDRMAtomic::{} - atomic commit failed: {}", __FUNCTION__,
               strerror(errno));
+    m_atomicRequestQueue.pop_back();
+  }
+  else if (m_atomicRequestQueue.size() > 1)
+  {
+    m_atomicRequestQueue.pop_front();
```

Comment:
In case the `DRM_MODE_ATOMIC_TEST_ONLY` commit failed and the fallback works, doesn't this remove the working request and keeps the bad one? Couldn't that become a problem?

Reply:
from the looks of it, you seem to be correct, however i remember i had stepped in those lines with a debuggger, and i remember it was tricky, i will run another verfication with a debugger and update.

The confusing part is in case of a test commit failure, it rolls back to previous buffer before queue pops are done. So there might be misjudgement of the back buffer element with the front buffer element due to previous swap.

Reply:
OK, lets keep it as is for now. I'm thinking about removing the queue anyway as it isn't really used like a queue it seems.

Reply:
exactly was my though, my mod seems to have issue but the original also had issues, tbh there is no need for a queue anyways all needed a ptr to last known good commit.